Analysis/src/BaseEventAnalyzer.cpp

```diff
@@ -305,11 +304,68 @@ void BaseEventAnalyzer::ProcessDataSource(const SampleDescriptor& sample, const
+                        static_cast<float>(lepton_weight_provider->GetTriggerWeight(*event,
+                            signalObjectSelector.GetTauVSjetDiscriminator().second, unc_source, unc_scale));
+
+                    if(unc_source == UncertaintySource::None) {
+                        uncs_weight_map[uncs_trigger_weight.at(unc)][UncertaintyScale::Up] =
+                            static_cast<float>(lepton_weight_provider->GetTriggerWeight(*event,
+                                signalObjectSelector.GetTauVSjetDiscriminator().second, uncs_trigger_weight.at(unc),
+                                UncertaintyScale::Up));
+                        uncs_weight_map[uncs_trigger_weight.at(unc)][UncertaintyScale::Down] =
+                            static_cast<float>(lepton_weight_provider->GetTriggerWeight(*event,
+                                signalObjectSelector.GetTauVSjetDiscriminator().second, uncs_trigger_weight.at(unc),
+                                UncertaintyScale::Down));
+                    }
+                }
+
+                for (size_t unc = 0; unc < uncs_tau_weight.size(); ++unc){
+                    uncs_weight_map[uncs_tau_weight.at(unc)][UncertaintyScale::Central] =
+                        static_cast<float>(lepton_weight_provider->GetIdIsoWeight(*event,
+                                signalObjectSelector.GetTauVSeDiscriminator(channelId).second,
+                                signalObjectSelector.GetTauVSmuDiscriminator(channelId).second,
+                                signalObjectSelector.GetTauVSjetDiscriminator().second, unc_source, unc_scale));
+
+                    if(unc_source == UncertaintySource::None) {
+                        uncs_weight_map[uncs_tau_weight.at(unc)][UncertaintyScale::Up] =
+                            static_cast<float>(lepton_weight_provider->GetIdIsoWeight(*event,
+                                    signalObjectSelector.GetTauVSeDiscriminator(channelId).second,
+                                    signalObjectSelector.GetTauVSmuDiscriminator(channelId).second,
+                                    signalObjectSelector.GetTauVSjetDiscriminator().second, uncs_tau_weight.at(unc),
+                                UncertaintyScale::Up));
+                        uncs_weight_map[uncs_tau_weight.at(unc)][UncertaintyScale::Down] =
+                            static_cast<float>(lepton_weight_provider->GetIdIsoWeight(*event,
+                                    signalObjectSelector.GetTauVSeDiscriminator(channelId).second,
+                                    signalObjectSelector.GetTauVSmuDiscriminator(channelId).second,
+                                    signalObjectSelector.GetTauVSjetDiscriminator().second, uncs_tau_weight.at(unc),
+                                UncertaintyScale::Down));
                     }
                 }
+                uncs_weight_map[UncertaintySource::TopPt][UncertaintyScale::Up] = static_cast<float>((*event)->weight_top_pt);
```

Comment:
- TopPt unc should be applied only for the top-related dataset (e.g. single top, ttbar and TTVV) => better to add flag in sources.cfg for the samples where top_pt unc should be applied.
- Store 1 for `UncertaintyScale::Central` - like this no need to have a special treatment when filling AnaTuple.
- `if` section related to `TT` in `ProcessSpecialEvent` is no longer relevant => remove it to avoid confusions and update sources.cfg to not treat `TT` as a "special".